TestClient_RestartTracker_ModeDelay fails on Windows.

When building commit: 2de95e087edeafa2053163ffe4de39414ea97236

go version go1.5.2 windows/amd64
Commands to build and test:

```
go get -t -d -u -v github.com/hashicorp/nomad/...
go install -v github.com/hashicorp/nomad/...
set PATH=%GOPATH%\bin;%PATH%
go test -p=1 -v github.com/hashicorp/nomad/...
```

Results:
```
<snip>
--- FAIL: TestClient_RestartTracker_ModeDelay (0.00s)
	restarts_test.go:48: NextRestart() returned 2m0s; want less than 2m0s and more than 1s
<snip>
```

--- a/client/restarts_test.go
+++ b/client/restarts_test.go
@@ -44,8 +44,8 @@ func TestClient_RestartTracker_ModeDelay(t *testing.T) {
 		if !actual {
 			t.Fail()
 		}
-		if !(when > p.Delay && when < p.Interval) {
-			t.Fatalf("NextRestart() returned %v; want less than %v and more than %v", when, p.Interval, p.Delay)
+		if !(when > p.Delay && when <= p.Interval) {
+			t.Fatalf("NextRestart() returned %v; want > %v and <= %v", when, p.Delay, p.Interval)
 		}
 	}
 }

